src/LinAlg/hiopVectorIntHip.cpp

```diff
@@ -63,52 +63,32 @@
 namespace hiop
 {
 
-
-
 hiopVectorIntHip::hiopVectorIntHip(size_type sz, std::string mem_space)
-  : hiopVectorInt(sz),
-    mem_space_(mem_space)
+  : hiopVectorInt(sz)
 {
-  // Size in bytes
-  size_t bytes = sz * sizeof(index_type);
-
-  // Allocate memory on GPU
-  hipError_t cuerr = hipMalloc(&buf_, bytes);
-  assert(hipSuccess == cuerr);
-  
-  // Allocate memory on host
-  buf_host_ = new index_type[bytes];
-}
-
-hiopVectorIntHip::~hiopVectorIntHip()
-{
-  delete buf_host_;
-
-  // Delete workspaces and handles
-  hipFree(buf_);
-}
-
-void hiopVectorIntHip::copy_from_dev()
-{
-  if (buf_ != buf_host_) {
-    hipError_t cuerr = hipMemcpy(buf_host_, buf_, (sz_)*sizeof(index_type), hipMemcpyDeviceToHost);
-    assert(cuerr == hipSuccess);
+  buf_ = exec_space_.template alloc_array<index_type>(sz);
+  if(exec_space_.mem_backend().is_device()) {
```

Comment:
I'm afraid you reviewed an old version. I fixed many things including this one in the next commits.
you'd better review the new changes again, since I need to change function signatures in `ExecSpace.hpp`.